### PR TITLE
Add better UX & UI for CourseItem.jsx in AdminCoursesDashboard.jsx

### LINKED ISSUE #55: Fix UX in admin dashboard - show all courses' info
For each course item in the courses list in the admid dashboard, need to display all the details about the course.
For the simplicity of the UI (which was the motivation for not including all of the information), it will be implemented with a "show more" / "show less" options.

### PR DESCRIPTION
**Pull request by Danielsio**
## Changes Summery
1. Added better description to buttons mainly in CourseItem.jsx
2. Added Modal for deletion
3. Added show more/less button toggle for course info

## Relevent Issues
close #55

### CODE REVIEW

File: src/components/CourseItem.jsx [hunk 1]
```diff
@@ -1,31 +1,118 @@
...
-  );
+
+function CourseItem({course}) {
+    const navigate = useNavigate();
+    const [showDetails, setShowDetails] = useState(false);
+    const [showDeleteModal, setShowDeleteModal] = useState(false);
+    const {subjectID} = useContext(UserContext);
+
+    const handleEditClick = () => {
+        console.log(course);
```
Comment (oshriFadlon1) [inline src/components/CourseItem.jsx:18]: delete print.

File: src/components/CourseItem.jsx [hunk 2]
```diff
@@ -1,31 +1,118 @@
...
+    const handleConfirmDelete = () => {
+        api
+            .delete(`/admin/courses/${course.courseId}`,
+                {
+                    params: {
+                        sub: subjectID
+                    }
+                })
+            .then(response => {
+                console.log(response)
```
Comment (oshriFadlon1) [inline src/components/CourseItem.jsx:35]: and here

**Review by oshriFadlon1 — COMMENTED** [review]
delete prints

**Comment by Danielsio** [comment]
> delete prints

done

**Review by oshriFadlon1 — APPROVED** [review]

File: src/components/CourseItem.jsx [hunk 3]
```diff
@@ -1,31 +1,114 @@
...
+                        <div>
+                            <Card.Text>Subjects:</Card.Text>
+                            <ul style={{listStyleType: "none", margin: 0, padding: 0}}>
+                                {course.courseSubjects.map((subject, index) => (
+                                    <li key={index}>{`${index + 1}. ${subject}`}</li>
+                                ))}
+                            </ul>
+                        </div>
+                    </div>
+                </Collapse>
```
Comment (itayf9) [inline src/components/CourseItem.jsx:80]: @Danielsio I think you should also add the "Subjects Practice Percentage" field that we have in a course's info.

File: src/pages/AdminCoursesDashboard.jsx [hunk 4]
```diff
@@ -132,7 +132,7 @@ function AdminCoursesDashboard() {
         </Col>
         <Col className="text-end add-course-btn-col">
           <Button variant="success" onClick={handleAddCourse}>
-            Add Course
+            Add New Course
```
Comment (itayf9) [inline src/pages/AdminCoursesDashboard.jsx:135]: @Danielsio In my opinion:
"Add _A_ New Course"
or
"New Course"

**Review by itayf9 — CHANGES_REQUESTED** [review]

File: src/components/CourseItem.jsx (hunk 3 above)
Comment (Danielsio) [reply src/components/CourseItem.jsx]: done

**Pull request merged**